pkg/backup/backup_compaction_test.go

```diff
@@ -119,6 +169,111 @@ func TestBackupCompaction(t *testing.T) {
 		require.Equal(t, numIndexes, restoredNumIndexes)
 	})
 
+	t.Run("compact middle of backup chain", func(t *testing.T) {
+		db.Exec(t, "CREATE TABLE foo (a INT, b INT)")
+		defer func() {
+			db.Exec(t, "DROP TABLE foo")
+		}()
+		db.Exec(t, "INSERT INTO foo VALUES (1, 1)")
+		db.Exec(t, "BACKUP INTO 'nodelocal://1/backup'")
+
+		db.Exec(t, "INSERT INTO foo VALUES (2, 2)")
+		start := getTime()
+		db.Exec(
+			t,
+			fmt.Sprintf("BACKUP INTO 'nodelocal://1/backup' AS OF SYSTEM TIME '%s'", start),
+		)
+
+		db.Exec(t, "INSERT INTO foo VALUES (3, 3)")
+		db.Exec(t, "BACKUP INTO LATEST IN 'nodelocal://1/backup'")
+
+		db.Exec(t, "INSERT INTO foo VALUES (4, 4)")
+		db.Exec(t, "BACKUP INTO LATEST IN 'nodelocal://1/backup'")
+
+		db.Exec(t, "INSERT INTO foo VALUES (5, 5)")
+		end := getTime()
+		db.Exec(
+			t,
+			fmt.Sprintf("BACKUP INTO LATEST IN 'nodelocal://1/backup' AS OF SYSTEM TIME '%s'", end),
+		)
+
+		db.Exec(t, "INSERT INTO foo VALUES (6, 6)")
+		db.Exec(t, "BACKUP INTO LATEST IN 'nodelocal://1/backup'")
+
+		db.Exec(t, fmt.Sprintf(compactionBuiltin, start, end))
+		validateCompactedBackupForTables(t, db, []string{"foo"}, "'nodelocal://1/backup'")
+	})
+
+	t.Run("table-level backups", func(t *testing.T) {
+		db.Exec(t, "CREATE TABLE foo (a INT, b INT)")
```

Comment:
did this catch a regression? I hope we can eventually delete this sort of test case after we sprinkle a backup compaction test hook throughout our backup-restore unit tests (i.e. right before running restore, maybe run a compaction with an endtime equal to the aost we're about to restore too). we can chat about this after this merges.

Reply:
It's not a regression, but it is for more coverage. There's a slightly different path when targets are specified, so wanted to make sure that was covered. For sure, once we sprinkle compactions across our other tests, then this can be removed.